compression.go

```diff
@@ -111,8 +115,12 @@ func (c *compressionPool) Compress(dst *bytes.Buffer, src *bytes.Buffer) *Error
 	if err != nil {
 		return errorf(CodeUnknown, "get compressor: %w", err)
 	}
-	if _, err := io.Copy(compressor, src); err != nil {
+	if _, err := src.WriteTo(compressor); err != nil {
```

Comment:
This was just for consistency/symmetry with the decompress code above, which directly calls `dst.ReadFrom` instead of using `io.Copy`.